orttraining/orttraining/test/training_api/core/checkpoint_test.cc

```diff
@@ -160,6 +160,36 @@ TEST(CheckpointApiTest, SaveOnnxModelAsCheckpoint_ThenLoad_CPU) {
   }
 }
 
+/**
+ * Load ONNX model with parameters set to 0 from file path, Load Checkpoint weights into the Model,
+ * Then compare the new weights to 0 to make sure they were changed after loading checkpoint to model.
+ */
+TEST(CheckpointApiTest, LoadCheckpointToModel) {
+  // Phase 1: Load a Model with weights set to zero.
+  auto model_uri = MODEL_FOLDER "transform/load_checkpoint/zero_model.onnx";
+  ONNX_NAMESPACE::ModelProto p_model;
+  ASSERT_STATUS_OK(Model::Load(model_uri, p_model));
+  // Phase 2: Load the checkpoint weights into the Model.
+  // Call Load APIs
+  PathString checkpoint_path = ORT_TSTR("testdata/transform/load_checkpoint");
+  ASSERT_STATUS_OK(LoadCheckpointToModel(checkpoint_path, p_model));
+
+  // Phase 3: Make sure the Model's weights are not equal to zero after loading the new ones.
+  // Load imported initializers into the Model
+  for (auto& init : p_model.graph().initializer()) {
+    // Convert the tensor bytes to a float vector to compare.
+    ORT_ENFORCE(init.has_raw_data(), "Raw data was not found on the Initializer.");
```

Comment:
in test code, prefer to use the gtest macros for consistency and better test output
```suggestion
    ASSERT_TRUE(init.has_raw_data());
```